social_protection/validation.py

```diff
@@ -7,12 +11,59 @@ class BenefitPlanValidation(BaseModelValidation):
 
     @classmethod
     def validate_create(cls, user, **data):
+        incoming_code = data.get('code')
+        code_error = check_bf_unique_code(incoming_code)
+        if code_error:
+            raise ValidationError(code_error[0]['message'])
+        incoming_name = data.get('name')
+        name_error = check_bf_unique_name(incoming_name)
+        if name_error:
+            raise ValidationError(name_error[0]['message'])
+        incoming_schema = data.get('beneficiary_data_schema')
+        schema_error = is_valid_json_schema(incoming_schema)
+        if schema_error:
+            raise ValidationError(schema_error[0]['message'])
         super().validate_create(user, **data)
 
     @classmethod
     def validate_update(cls, user, **data):
+        uuid = data.get('id')
+        incoming_code = data.get('code')
+        code_error = check_bf_unique_code(incoming_code, uuid)
+        if code_error:
+            raise ValidationError(code_error[0]['message'])
+        incoming_name = data.get('name')
+        name_error = check_bf_unique_name(incoming_name, uuid)
+        if name_error:
+            raise ValidationError(name_error[0]['message'])
+        incoming_schema = data.get('beneficiary_data_schema')
+        schema_error = is_valid_json_schema(incoming_schema)
+        if schema_error:
+            raise ValidationError(schema_error[0]['message'])
```

Comment:
Please introduce changed from create validation in here as well. 
Would it be possible to extract checking functionality to protected method? in both cases they're very similar

Reply:
done